### PR TITLE
[BARO] Add DPS310 barometer driver; enable on some targets

### PR DESCRIPTION
**Pull request by digitalentity**
Support for a very accurate barometer from Infineon. A drop-in replacement from BMP280.

### CODE REVIEW

**Comment by digitalentity** [comment]
Waiting for breakout  board to arrive.

**Comment by jflyper** [comment]
@digitalentity So, this is not a working code yet, is it?

**Comment by digitalentity** [comment]
@jflyper not yet. Waiting for a dev board to arrive so I can test this

**Comment by MATEKSYS** [comment]
image

**Comment by digitalentity** [comment]
Fixed, tested, looks good.

**Comment by jflyper** [comment]
Hmm... I see that oversample/precision is set to 16/standard.

Will this affect how pressure/altitude is reported to barometer module?
I'm asking because in a Betaflight port of this driver, I see altitude change in 20~23cm steps.

File: src/main/drivers/barometer/barometer_dps310.c [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    uint8_t buf[6];
+    if (!busReadBuf(baro->busDev, DPS310_REG_PSR_B2, buf, 6)) {
+        return false;
+    }
+
+    const int32_t Praw = getTwosComplement((buf[0] << 16) + (buf[1] << 8) + buf[2], 24);
+    const int32_t Traw = getTwosComplement((buf[3] << 16) + (buf[4] << 8) + buf[5], 24);
+
+    // 4. Calculate scaled measurement results.
+    const float Praw_sc = Praw / kP;
```
Comment (jflyper) [inline src/main/drivers/barometer/barometer_dps310.c:233]: I think this is causing the digit loss.

Comment (jflyper) [reply src/main/drivers/barometer/barometer_dps310.c:233]: Hmm... Still stepping ...
Betaflight_Configurator

**Comment by jflyper** [comment]
All solved. Betaflight `baroCalculateAltitude` was using some ancient code from ArduPilot that caused round up (down)s. Borrowed iNav's `pressureToAltitude` and all is good.

**Pull request merged**